snapshots/overlay/overlayutils/check.go

```diff
@@ -168,3 +174,15 @@ func NeedsUserXAttr(d string) (bool, error) {
 	}
 	return true, nil
 }
+
+// SupportsIndex checks whether the "index=off" option is supported by the kernel.
+func SupportsIndex() bool {
```

Comment:
Storing the result in a global variable isn't necessary. This is only called once when NewSnapshotter is called or during tests.

Reply:
done

Reply:
you mean it is not necessary to use Singleton Pattern? if i use sync.Once i have to use global variable
```
var (
	supportsIndex bool
	once          sync.Once
)
```

Reply:
I refered to https://github.com/containerd/containerd/blob/717dde3c27145dab133a35250434b270935f834f/pkg/userns/userns_linux.go#L26-L34